lidofinance/trp-ui
# Delegate snapshot voting power

## Description

- Refactored a bunch of components to be reusable.
- Added a page with snapshot voting power delegation. 
- Footer is the same as on staking widget.
- Header is the same as on staking widget.

## Demo

Main page:

<img width="1512" alt="image" src="https://user-images.githubusercontent.com/103929444/232567362-8f278d88-e975-43e8-82f3-da444615d684.png">

Snapshot page:

<img width="1512" alt="image" src="https://user-images.githubusercontent.com/103929444/232567421-ae2ef440-7a7a-4983-b56b-227dd9b48150.png">

## Review notes

- Looks like I accidentally forked from `main` so there are changelog.md changes. 
- Changes in .github folder are just because of prettier run, needed it to format new abi file.
- During components refactoring, moved to more or less the same structure we have on most apps: `features` for business logic, `shared/{api,lib,ui}` for reusable files without any business logic.
- Added [react-hook-form](https://react-hook-form.com/) for snapshot form.

## Testing notes

- Need to test that claim still works.
- Need to check that snapshot vote power delegation works.

btw we discussed it with @Jabher, that we won't automatically set current address into delegate field, so it's ok.

How to test:
1. Use web app to delegate to any address, wait for success status message.
2. Go to https://goerli.etherscan.io/address/0x469788fE6E9E9681C6ebF3bF78e7Fd26Fc015446#events
  - For mainnet it's https://etherscan.io/address/0x469788fE6E9E9681C6ebF3bF78e7Fd26Fc015446#events
4. If you see your address there (without leading 0x), then everything is ok
<img width="1512" alt="image" src="https://user-images.githubusercontent.com/103929444/232758426-f4261d41-089e-4a66-943e-e38efa7b4adb.png">
